src/main/java/com/redhat/devtools/lsp4ij/features/semanticTokens/SemanticTokensData.java

```diff
@@ -60,10 +61,12 @@ public SemanticTokens getSemanticTokens() {
      * @param document the document.
      * @param addInfo  callback to collect {@link HighlightInfo} created from the semantic tokens data.
      */
-    @Nullable
     public void highlight(@NotNull PsiFile file,
                           @NotNull Document document,
                           @NotNull Consumer<HighlightInfo> addInfo) {
+        // Try to populate the file's view provider with these tokens if possible

```

Comment:
This just piggybacks on the existing semantic tokens processing to add tokens to a file if it's the right type and the feature is enabled.